### PR TITLE
[AIRFLOW-3677] Improve check operator module test coverage

### PR DESCRIPTION
**Pull request by feluelle**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3677
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

This PR refactors **CheckOperator**, **ValueCheckOperator** and **IntervalCheckOperator** code and adds tests (see below)

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

- add missing tests cases for CheckOperator
- add missing tests cases for ValueCheckOperator

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.
  - All the public functions and the classes in the PR contain docstrings that explain what it does

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

**Comment by feluelle** [comment]
PTAL @ashb @Fokko @feng-tao @XD-DENG when you got time :)

File: airflow/operators/check_operator.py [hunk 1]
```diff
@@ -215,8 +223,7 @@ def __init__(
         self.days_back = -abs(days_back)
         self.conn_id = conn_id
         sqlexp = ', '.join(self.metrics_sorted)
-        sqlt = ("SELECT {sqlexp} FROM {table}"
-                " WHERE {date_filter_column}=").format(**locals())
+        sqlt = "SELECT {sqlexp} FROM {table} WHERE {date_filter_column}=".format(**locals())
```
Comment (Fokko) [inline airflow/operators/check_operator.py]: Can we get rid of `**locals()` here and be explicit?

Comment (feluelle) [reply airflow/operators/check_operator.py]: Sure. :)

**Comment by feluelle** [comment]
Due to changes in #4983 I removed some parts that have already been added.

Please @Fokko @feng-tao @XD-DENG review this once again so we finally can get this in.

**Review by Fokko — APPROVED** [review]
LGTM, thanks @feluelle

**Pull request merged**